src/main/java/com/crafter/springboottechiteasy/controllers/TelevisionController.java

```diff
@@ -2,56 +2,54 @@
 
 import com.crafter.springboottechiteasy.exceptions.RecordNotFoundException;
 import com.crafter.springboottechiteasy.exceptions.RequirementsNotMetException;
+import com.crafter.springboottechiteasy.models.Television;
+import com.crafter.springboottechiteasy.repositories.TelevisionRepository;
+import com.crafter.springboottechiteasy.services.TelevisionService;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("/televisions")
 public class TelevisionController {
 
+    private TelevisionService televisionService;
+
+    public TelevisionController(TelevisionService televisionService) {
+        this.televisionService = televisionService;
+    }
+
     @GetMapping
-    public ResponseEntity<Object> getAllTVs() {
-        return ResponseEntity.ok("television");
+    public ResponseEntity<List<Television>> getAllTVs() {
+        return ResponseEntity.ok(televisionService.getAllTelevisions());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Object> getOneTV(@PathVariable int id) {
-        if (id == 0) { // determine if id exists
-            throw new IndexOutOfBoundsException();
-        } else {
-            return ResponseEntity.ok(id);
-        }
+    public ResponseEntity<Television> getOneTV(@PathVariable Long id) {
+        return ResponseEntity.ok(televisionService.getOneTelevision(id));
     }
 
     @PostMapping("/add")
-    public ResponseEntity<Object> createNewTV() {
-        String television = "Panasonic";
-        if (television.length() > 20) {
-            throw new RequirementsNotMetException("The name is too long. There is a max of 20 characters");
-        } else {
-            return ResponseEntity.created(null).body("television");
-        }
+    public ResponseEntity<Television> createNewTV(@RequestBody Television t) {
+        return ResponseEntity.created(null).body(televisionService.createTelevision(t));
```

Comment:
Probeer voor de volgende opdracht of je hier de null in .created() kunt vervangen door de URI die lijdt naar de get request voor de zojuist aangemaakte television. Dat wordt namelijk de "Location" header van jou http response. REST verwacht een location header bij de response van een POST request.